state-sync/state-sync-v2/state-sync-driver/src/bootstrapper.rs

```diff
@@ -629,71 +613,49 @@ impl<MempoolNotifier: MempoolNotificationSender, StorageSyncer: StorageSynchroni
-                            notification_id,
-                            NotificationFeedback::PayloadTypeIsIncorrect,
-                        )
-                        .await?;
-                        return Err(Error::InvalidPayload(
-                            "Did not receive transaction outputs with proof!".into(),
-                        ));
-                    }
-                }
-                BootstrappingMode::ExecuteTransactionsFromGenesis => {
-                    if let Some(transaction_list_with_proof) = transaction_list_with_proof {
-                        let committed_transactions =
-                            transaction_list_with_proof.transactions.clone();
-                        let committed_events = self
-                            .storage_synchronizer
-                            .lock()
-                            .execute_and_commit_transactions(
-                                transaction_list_with_proof,
-                                highest_known_ledger_info,
-                                end_of_epoch_ledger_info,
-                            );
-                        (committed_events, committed_transactions)
-                    } else {
-                        self.terminate_active_stream(
-                            notification_id,
-                            NotificationFeedback::PayloadTypeIsIncorrect,
-                        )
-                        .await?;
-                        return Err(Error::InvalidPayload(
-                            "Did not receive transactions with proof!".into(),
-                        ));
-                    }
+        match self.driver_configuration.config.bootstrapping_mode {
+            BootstrappingMode::ApplyTransactionOutputsFromGenesis => {
+                if let Some(transaction_outputs_with_proof) = transaction_outputs_with_proof {
+                    self.storage_synchronizer.lock().apply_transaction_outputs(
+                        transaction_outputs_with_proof,
+                        highest_known_ledger_info,
+                        end_of_epoch_ledger_info,
+                    )?;
+                } else {
+                    self.terminate_active_stream(
+                        notification_id,
+                        NotificationFeedback::PayloadTypeIsIncorrect,
+                    )
+                    .await?;
+                    return Err(Error::InvalidPayload(
+                        "Did not receive transaction outputs with proof!".into(),
+                    ));
                 }
-                bootstrapping_mode => {
-                    unimplemented!("Bootstrapping mode not supported: {:?}", bootstrapping_mode)
+            }
+            BootstrappingMode::ExecuteTransactionsFromGenesis => {
+                if let Some(transaction_list_with_proof) = transaction_list_with_proof {
+                    self.storage_synchronizer.lock().execute_transactions(
```

Comment:
The lock seems too big? Is it possible that other threads get blocked on this?

Reply:
scratch that ;D

Reply:
Yeah, this should only be used by a single thread (the driver/bootstrapper/continuous syncer) after the last commit, so hopefully no blocking 😄

Reply:
FYI: I'll be removing this lock entirely in the next PR.